buildSrc/src/main/kotlin/com/wire/kalium/plugins/Multiplatform.kt

```diff
@@ -37,6 +37,9 @@ fun Project.configureDefaultMultiplatform(
             //       this should require us moving iOS code from iOSX64Main to
             //       another sourceSet (iOSMain/iOSCommon or similar)
             iosX64()
+            iosArm64()
+            iosSimulatorArm64()
+            macosX64()
```

Comment:
I don't know why the `getting` and `creating` delegates are not working in this file, at least the IDE doesn't like it but it seems to work.

We can get away by not using the sugar syntax and going a bit more primitive here:

```suggestion
            macosX64()
            
            val commonMain = sourceSets.getByName("commonMain")
            val commonTest = sourceSets.getByName("commonTest")
            val darwinMain = sourceSets.create("darwinMain") {
                dependsOn(commonMain)
            }
            val darwinTest = sourceSets.create("darwinTest") {
                dependsOn(commonTest)
            }
            val iosX64Main = sourceSets.getByName("iosX64Main") {
                dependsOn(darwinMain)
            }
            val iosX64Test = sourceSets.getByName("iosX64Test") {
                dependsOn(darwinTest)
            }
            val iosArm64Main = sourceSets.getByName("iosArm64Main") {
                dependsOn(darwinMain)
            }
            val iosArm64Test = sourceSets.getByName("iosArm64Test") {
                dependsOn(darwinTest)
            }
            val iosSimulatorArm64Main = sourceSets.getByName("iosSimulatorArm64Main") {
                dependsOn(darwinMain)
            }
            val iosSimulatorArm64Test = sourceSets.getByName("iosSimulatorArm64Test") {
                dependsOn(darwinTest)
            }
            val macosX64Main = sourceSets.getByName("macosX64Main") {
                dependsOn(darwinMain)
            }
            val macosX64Test = sourceSets.getByName("macosX64Test") {
                dependsOn(darwinTest)
            }
```

Maybe extracting this block to a function would be better:

```kotlin
fun KotlinMultiplatformExtension.darwinMultiplatformConfig() {
    val commonMain = sourceSets.getByName("commonMain")
    val commonTest = sourceSets.getByName("commonTest")
    val darwinMain = sourceSets.create("darwinMain") {
        dependsOn(commonMain)
    }
    val darwinTest = sourceSets.create("darwinTest") {
        dependsOn(commonTest)
    }
    val iosX64Main = sourceSets.getByName("iosX64Main") {
        dependsOn(darwinMain)
    }
    val iosX64Test = sourceSets.getByName("iosX64Test") {
        dependsOn(darwinTest)
    }
    val iosArm64Main = sourceSets.getByName("iosArm64Main") {
        dependsOn(darwinMain)
    }
    val iosArm64Test = sourceSets.getByName("iosArm64Test") {
        dependsOn(darwinTest)
    }
    val iosSimulatorArm64Main = sourceSets.getByName("iosSimulatorArm64Main") {
        dependsOn(darwinMain)
    }
    val iosSimulatorArm64Test = sourceSets.getByName("iosSimulatorArm64Test") {
        dependsOn(darwinTest)
    }
    val macosX64Main = sourceSets.getByName("macosX64Main") {
        dependsOn(darwinMain)
    }
    val macosX64Test = sourceSets.getByName("macosX64Test") {
        dependsOn(darwinTest)
    }
}

```

Reply:
Updated